### PR TITLE
remove deprecated --dynamic-whitelist option

### PR DESCRIPTION
**Pull request by hroff-1902**
Removal of the `--dynamic-whitelist` option, which left in the deprecated status for long time.

That was not so easy (many appearances), so verify that I did not forget anything.

### CODE REVIEW

**Comment by hroff-1902** [comment]
Amongst other changes, I removed "If enabled, it overrides the dynamic whitelist option." in docs/edge.md, since it seems for me incorrect and confusing. Edge does overrides whitelist options, it modifies the whitelist by itself, right?

**Comment by hroff-1902** [comment]
In deprecated.md I stated "This command line option was deprecated in 2018 and removed freqtrade 2019.6-dev (develop branch) and in freqtrade 2019.7 (master branch)."

Version from the master branch should be adjusted during release process.

File: docs/configuration.md [hunk 1]
```diff
@@ -44,17 +44,17 @@ Mandatory Parameters are marked as **Required**.
...
 | `exchange.markets_refresh_interval` | 60 | The interval in minutes in which markets are reloaded.
 | `edge` | false | Please refer to [edge configuration document](edge.md) for detailed explanation.
 | `experimental.use_sell_signal` | false | Use your sell strategy in addition of the `minimal_roi`. [Strategy Override](#parameters-in-the-strategy).
 | `experimental.sell_profit_only` | false | Waits until you have made a positive profit before taking a sell decision. [Strategy Override](#parameters-in-the-strategy).
 | `experimental.ignore_roi_if_buy_signal` | false | Does not sell if the buy-signal is still active. Takes preference over `minimal_roi` and `use_sell_signal`. [Strategy Override](#parameters-in-the-strategy).
-| `pairlist.method` | StaticPairList | Use Static whitelist. [More information below](#dynamic-pairlists).
-| `pairlist.config` | None | Additional configuration for dynamic pairlists. [More information below](#dynamic-pairlists).
+| `pairlist.method` | StaticPairList | Use static or dynamic volume-based pairlist.
+| `pairlist.config` | None | Additional configuration for dynamic pairlists.
```
Comment (xmatthias) [inline docs/configuration.md]: Any particular reason to remove the link to the documentation (which is still there ...?)

File: docs/configuration.md [hunk 2]
```diff
@@ -44,17 +44,17 @@ Mandatory Parameters are marked as **Required**.
 | `exchange.sandbox` | false | Use the 'sandbox' version of the exchange, where the exchange provides a sandbox for risk-free integration. See [here](sandbox-testing.md) in more details.
 | `exchange.key` | '' | API key to use for the exchange. Only required when you are in production mode.
 | `exchange.secret` | '' | API secret to use for the exchange. Only required when you are in production mode.
-| `exchange.pair_whitelist` | [] | List of currency to use by the bot. Can be overrided with `--dynamic-whitelist` param.
-| `exchange.pair_blacklist` | [] | List of currency the bot must avoid. Useful when using `--dynamic-whitelist` param.
+| `exchange.pair_whitelist` | [] | List of currency to use by the bot.
```
Comment (xmatthias) [inline docs/configuration.md]: I guess we should still mention that `pairlist.method=dynamic` will override this?

File: docs/configuration.md [hunk 3]
```diff
@@ -381,7 +381,7 @@ section of the configuration.
   * It uses configuration from `exchange.pair_whitelist` and `exchange.pair_blacklist`.
 * `VolumePairList`
   * Formerly available as `--dynamic-whitelist [<number_assets>]`. This command line
-option is deprecated and should no longer be used.
+option was removed and can no longer be used.
```
Comment (xmatthias) [inline docs/configuration.md]: i would just remove it from the documentation - no point in mentioning that we removed it (it'll only confuse new users)

File: docs/configuration.md (hunk 2 above)
Comment (hroff-1902) [reply docs/configuration.md]: Oops, I thought it was a link to a section with description of the deprecated option. Restored it.

File: docs/configuration.md (hunk 1 above)
Comment (hroff-1902) [reply docs/configuration.md]: Oops, I thought it was a link to a section with description of the deprecated option. Restored it.

File: docs/configuration.md (hunk 2 above)
Comment (hroff-1902) [reply docs/configuration.md]: I have a better idea I guess. See below or in a sep. issue describing the proposal (or a PR with implementation).

File: docs/configuration.md (hunk 3 above)
Comment (hroff-1902) [reply docs/configuration.md]: Agree, removed.

**Comment by hroff-1902** [comment]
The idea regarding whitelist I mentioned above is to combine them: so that VolumePairList select pairs from the set of [Whitelist - Blacklist], not from all pairs available. So the white list won't be overriden by VolumePairList or ignored, but will be used as a starting point for this method.

Rationale: a user could implement selecting top 5 pairs by its volume from a set of 10 pairs she is interesting to trade. You may say that even currently she may specify all non-interesting pairs in the blacklist. Well, yes, but 1) she needs to follow exchange and add quickly all new shitcoins listed in the exchange, 2) if exchange has 437 pairs and a user wants to trade only 10 or 20 of them, the blacklist will be extremely long. With my proposal she will be able to specify the list on any manner, choosing the most convenient approach (and use VolumePairList method on the resulting [Whitelist - Blacklist].

This can be even more useful in the future with addition more dynamic pairlist methods.

**Comment by xmatthias** [comment]
yes and no - it depends on the usecase (i may not want to blacklist any pairs) - so your approach generates some work to generate a whitelist i currently simply leave empty (Yes, that's how i do it ...)

Why not add a config to VolumePairList to "selectFromWhitelist":True to activate (or default to this) behaviour?

File: docs/configuration.md (hunk 2 above)
Comment (xmatthias) [reply docs/configuration.md]: apart from that (see below) - both this and the next line are oddly worded... (i've just seen that now)

we have pairs in this list, not currencies ... 
maybe 
`List of pairs to check for potential trades.`
`List of pairs to absolutely avoid.`
?

**Comment by xmatthias** [comment]
well - it sounds error-prone (assume i mistype "pair_white1ist" ) ...

It'll produce unexpected results ... and i'm not sure i like it (it's not for this PR anyway ...) so i guess this should better be discussed in an issue (or the PR implementing this feature)...

File: docs/configuration.md (hunk 2 above)
Comment (hroff-1902) [reply docs/configuration.md]: Yes, right.

**Comment by hroff-1902** [comment]
> it's not for this PR anyway ...) so i guess this should better be discussed in an issue (or the PR implementing this feature)...

Sure.

File: docs/configuration.md (hunk 2 above)
Comment (hroff-1902) [reply docs/configuration.md]: But not 'for potential trades' since whitelist is used in live trades as we know... ;-)

Comment (hroff-1902) [reply docs/configuration.md]: Fixed, pls review.

**Pull request merged**